packages/utilities/psammead-calendars/src/calendars/jalaali.js

```diff
@@ -0,0 +1,61 @@
+const jalaaliJs = require('jalaali-js');
+
+const jalaaliMonths = {
+  fa: [
+    'فروردین',
+    'اردیبهشت',
+    'خرداد',
+    'تیر',
+    'مرداد',
+    'شهریور',
+    'مهر',
+    'آبان',
+    'آذر',
+    'دی',
+    'بهمن',
+    'اسفند',
+  ],
+  ps: [
+    'وری',
+    'غویی',
+    'غبرګولی',
+    'چنګاښ',
+    'زمری',
+    'وږی',
+    'تله',
+    'لړم',
+    'لیندۍ',
+    'مرغومی',
+    'سلواغه',
+    'کب',
+  ],
+};
+
+const jalaali = {
+  formatDate: gregorianMoment => {
+    // Type checks for passed in value gregorianMoment
+    if (
+      !gregorianMoment ||
+      typeof gregorianMoment.locale !== 'function' ||
+      !gregorianMoment.isValid()
+    ) {
+      return null;
+    }
+    const currentLocale = gregorianMoment.locale();
+    // Check if locale passed in is valid
+    if (!(`${currentLocale}` in jalaaliMonths)) {
+      return null;
+    }
+    const jalaaliDate = jalaaliJs.toJalaali(
+      gregorianMoment.year(),
+      gregorianMoment.month() + 1,
+      gregorianMoment.date(),
+    );
+    const localeJalaaliMonths = jalaaliMonths[gregorianMoment.locale()];
+    const jalaaliMonth = localeJalaaliMonths[jalaaliDate.jm - 1];
+    const output = `${jalaaliDate.jd} ${jalaaliMonth} ${jalaaliDate.jy}`;
+    return output;
+  },
```

Comment:
This function is getting quite long, what about splitting out the validation into a separate internal `isSupportedMoment` helper function and leaving the rest of this function just carrying out the actual conversion/formatting? This would help make the code easier to follow.

Reply:
Yeah that makes sense. Something like this? 

```
// Helper function to check if argument passed through is a supported moment
const isSupportedMoment = moment => {
  if (
    moment &&
    typeof moment.locale === 'function' &&
    moment.isValid() &&
    `${moment.locale()}` in jalaaliMonths
  ) {
    return true;
  }
  return false;
};
```
Which would make the formatting look like this:
```
const jalaali = {
  formatDate: gregorianMoment => {
    if (isSupportedMoment(gregorianMoment)) {
      const jalaaliDate = jalaaliJs.toJalaali(
        gregorianMoment.year(),
        gregorianMoment.month() + 1,
        gregorianMoment.date(),
      );
      const localeJalaaliMonths = jalaaliMonths[gregorianMoment.locale()];
      const jalaaliMonth = localeJalaaliMonths[jalaaliDate.jm - 1];
      const output = `${jalaaliDate.jd} ${jalaaliMonth} ${jalaaliDate.jy}`;
      return output;
    }
    return null;
  },
};
```